### PR TITLE
Generic multi-component mesh 

### PR DESCRIPTION
**Pull request by brownbaerchen**
As discussed in #399, @tlunet, @lisawim and I have been working on a generic multi-component mesh implementation.
This can definitely still be fleshed out somewhat, but is already an improvement over the current state, especially with respect to @lisawim's current project.
She needs a mesh like this that can be plugged into a scipy solver, but that also allows to reference the contents somewhat meaningfully.
In the future, we should extend this to other parts of the code. But let's merge one feature at a time.

### CODE REVIEW

File: pySDC/implementations/datatype_classes/mesh.py [hunk 1]
```diff
@@ -150,6 +150,41 @@ def bcast(self, root=None, comm=None):
 
 
 class MultiComponentMesh(mesh):
+    """
```
[the remark this answers is not in the document]
Comment (tlunet) [reply pySDC/implementations/datatype_classes/mesh.py]: ah ... @brownbaerchen did the classical rooky mistakes :stuck_out_tongue_closed_eyes:

[the remark this answers is not in the document]
Comment (brownbaerchen) [reply pySDC/implementations/datatype_classes/mesh.py]: Well, your fancy math formatting is just distracting in my vim, so I overlook it as best I can :D

[the remark this answers is not in the document]
Comment (tlunet) [reply pySDC/implementations/datatype_classes/mesh.py]: What ?!? vim cannot render restructured text ???

**Comment by pancetta** [comment]
All happy?

**Pull request merged**